src/benchq/mlflow/data_logging.py

```diff
@@ -0,0 +1,68 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+""" Tools for using mlflow to log benchq data """
+from dataclasses import asdict
+from numbers import Number
+from typing import Any, Dict, Optional
+
+import mlflow  # type: ignore
+
+from ..data_structures import (
+    AlgorithmImplementation,
+    BasicArchitectureModel,
+    DecoderModel,
+    ResourceInfo,
+)
+
+
+def log_input_objects_to_mlflow(
+    algorithm_description: AlgorithmImplementation,
+    algorithm_name: str,
+    hardware_model: BasicArchitectureModel,
+    decoder_model: Optional[DecoderModel] = None,
+):
+    """Ingests objects used to define a resource estimation
+    and logs their parameters to mlflow
+    """
+    # Sometimes the algorithm description has a None as a value,
+    #   which causes problem if we try to log it
+    for algo_key, algo_value in _flatten_dict(asdict(algorithm_description)).items():
+        if algo_value is not None:
+            mlflow.log_param(algo_key, algo_value)
+        else:
+            mlflow.log_param(algo_key, "None")
+
+    mlflow.log_param("algorithm_name", algorithm_name)
+
+    mlflow.log_params(_flatten_dict(asdict(hardware_model)))
+
+    # because decoder model is optional, check to make sure it exists
+    if decoder_model is not None:
+        mlflow.log_params(_flatten_dict(asdict(decoder_model)))
+    else:
+        mlflow.log_param("decoder_model", "None")
+
+
+def log_resource_info_to_mlflow(resource_info: ResourceInfo):
```

Comment:
Add `-> None` to indicate return value type

Reply:
fixed in ba87b5c7cf219232747b7e560dae8ec237e610c5